src/univariate/continuous/lindley.jl

```diff
@@ -0,0 +1,184 @@
+
+# Derived with Mathematica:
+#     KLDivergence := ResourceFunction["KullbackLeiblerDivergence"]
+#     KLDivergence[LindleyDistribution[θp], LindleyDistribution[θq]]
+function kldivergence(p::Lindley, q::Lindley)
+    θp = shape(p)
+    θq = shape(q)
+    a = (θp + 2) * (θp - θq) / θp / (1 + θp)
+    b = 2 * log(θp) + log1p(θq) - 2 * log(θq) - log1p(θp)
+    return b - a
+end
+
+# Derived with Mathematica based on https://mathematica.stackexchange.com/a/275765:
+#     ShannonEntropy[dist_?DistributionParameterQ] :=
+#         Expectation[-LogLikelihood[dist, {x}], Distributed[x, dist]]
+#     Simplify[ShannonEntropy[LindleyDistribution[θ]]]
+function entropy(d::Lindley)
+    θ = shape(d)
+    return 1 + exp(θ) * expinti(-θ) / (1 + θ) - 2 * log(θ) + log1p(θ)
+end
+
+### Evaluation
+
+_lindley_mgf(θ, t) = θ^2 * (1 + θ - t) / (1 + θ) / (θ - t)^2
+
+mgf(d::Lindley, t::Real) = _lindley_mgf(shape(d), t)
+
+cf(d::Lindley, t::Real) = _lindley_mgf(shape(d), t * im)
+
+cgf(d::Lindley, t::Real) = log1p(-t / (1 + d.θ)) - 2 * log1p(-t / d.θ)
+
+_zero(d::Lindley, y::Real) = zero(shape(d)) * zero(y)
+_oftype(d::Lindley, y::Real, x) = oftype(_zero(d, y), x)
+
+function pdf(d::Lindley, y::Real)
+    θ = shape(d)
+    if isnan(y)
+        return _oftype(d, y, NaN)
+    elseif isfinite(y) && y > 0
+        return θ^2 / (1 + θ) * (1 + y) * exp(-θ * y)
+    else
+        return _zero(d, y)
+    end
+end
+
+function logpdf(d::Lindley, y::Real)
+    θ = shape(d)
+    if isnan(y)
+        return _oftype(d, y, NaN)
+    elseif isfinite(y) && y > 0
+        return 2 * log(θ) - log1p(θ) + log1p(y) - θ * y
+    else
+        return _oftype(d, y, -Inf)
+    end
+end
+
+function gradlogpdf(d::Lindley, y::Real)
+    if isnan(y)
+        return _oftype(d, y, NaN)
+    elseif isfinite(y) && y > 0
+        return inv(1 + y) - shape(d)
+    else
+        return _zero(d, y)
+    end
+end
+
+function ccdf(d::Lindley, y::Real)
+    θ = shape(d)
+    θy = θ * y
+    if isnan(y)
+        return _oftype(d, y, NaN)
+    elseif y > 0
+        if isfinite(y)
+            return (1 + θy / (1 + θ)) * exp(-θy)
+        else
+            return _zero(d, y)
+        end
+    else
+        return _oftype(d, y, 1)
+    end
+end
+
+function logccdf(d::Lindley, y::Real)
+    θ = shape(d)
+    if isnan(y)
+        return _oftype(d, y, NaN)
+    elseif y > 0
+        if isfinite(y)
+            return log1p(θ * (1 + y)) - log1p(θ) - θ * y
+        else
+            return _oftype(d, y, -Inf)
+        end
+    else
+        return _zero(d, y)
+    end
+end
+
+cdf(d::Lindley, y::Real) = 1 - ccdf(d, y)
+
+logcdf(d::Lindley, y::Real) = log1p(-ccdf(d, y))
+
+# Jodrá, P. (2010). Computer generation of random variables with Lindley or
+# Poisson–Lindley distribution via the Lambert W function. Mathematics and Computers
+# in Simulation, 81(4), 851–859.
+function quantile(d::Lindley, q::Real)
+    θ = shape(d)
+    return -1 - inv(θ) - lambertw((1 + θ) * (q - 1) / exp(1 + θ), -1) / θ
```

Comment:
Could avoid one division:
```suggestion
    return -(1 + (1 + lambertw((1 + θ) * (q - 1) / exp(1 + θ), -1)) / θ)
```